core/services/workflows/syncer/handler.go

```diff
@@ -586,6 +585,226 @@ func (h *eventHandler) tryEngineCleanup(wfID string) error {
 	return nil
 }
 
+// registerWorkflow handles creating a workflow from contract workflow metadata. This method must remain idempotent.
+func (h *eventHandler) registerWorkflow(
```

Comment:
How come we're not using the other registerWorkflow handler?

Reply:
I think that's fair. There are some differences like the checkpointing. I made a separate handler to ensure that the default of the feature flag doesn't have changed behavior.